webapp/channels/src/components/suggestion/suggestion_box/suggestion_box.test.tsx

```diff
@@ -0,0 +1,244 @@
+
+        await waitFor(() => {
+            // Note that debouncing causes the provider to only be called once when the user stops typing
+            expect(providerSpy).toHaveBeenCalledTimes(1);
+        });
+
+        expect(screen.queryByRole('list')).toBeVisible();
+
+        expect(screen.queryByRole('list')).toBeVisible();
+        expect(screen.getByText('Suggestion: testtest')).toBeVisible();
+
+        // Typing more text should cause the suggestion to be updaetd
+        await userEvent.keyboard('words');
+
+        await waitFor(() => {
+            expect(providerSpy).toHaveBeenCalledTimes(2);
+        });
+
+        expect(screen.queryByRole('list')).toBeVisible();
+        expect(screen.getByText('Suggestion: testwordstestwords')).toBeVisible();
+
+        // Clearing the textbox hides all suggestions
+        await userEvent.clear(screen.getByPlaceholderText('test input'));
+
+        expect(screen.queryByRole('list')).not.toBeInTheDocument();
+    });
+
+    test('should hide suggestions on pressing escape', async () => {
+        const provider = new TestProvider();
+
+        renderWithContext(
+            <TestWrapper
+                {...makeBaseProps()}
+                providers={[provider]}
+            />,
+        );
+
+        // Start with no suggestions rendered
+        expect(screen.queryByRole('list')).not.toBeInTheDocument();
+
+        // Typing some text should cause a suggestion to be shown
+        userEvent.click(screen.getByPlaceholderText('test input'));
+        await userEvent.keyboard('test');
+
+        await waitFor(() => {
+            expect(screen.getByRole('list')).toBeVisible();
+        });
+
+        // Pressing escape hides all suggestions
+        await userEvent.keyboard('{escape}');
+
+        expect(screen.queryByRole('list')).not.toBeInTheDocument();
+    });
+
+    test('should autocomplete suggestions by pressing enter', async () => {
+        const provider = new TestProvider();
+
+        renderWithContext(
+            <TestWrapper
+                {...makeBaseProps()}
+                providers={[provider]}
+            />,
+        );
+
+        // Typing some text should cause a suggestion to be shown
+        userEvent.click(screen.getByPlaceholderText('test input'));
+        await userEvent.keyboard('test');
+
+        await waitFor(() => {
+            expect(screen.queryByRole('list')).toBeVisible();
+            expect(screen.getByText('Suggestion: testtest')).toBeVisible();
+        });
+
+        // Pressing enter should update the textbox value and hide the suggestion list
+        await userEvent.keyboard('{enter}');
+
+        await waitFor(() => {
+            expect(screen.getByPlaceholderText('test input')).toHaveValue('testtest ');
+        });
+
+        expect(screen.queryByRole('list')).not.toBeInTheDocument();
+    });
+
+    test('MM-57320 completing text with enter and calling resultCallback twice should not erase text following caret', async () => {
```

Comment:
I've been looking at this test code for far too long, so feel free to tell me if anything looks bizarre here. This component is super convoluted, and trying to cause Testing Library to trigger the bug is a mess